### DISCUSSION TITLE
Is it possible to extend the Daisy components like in Tailwind e.g. for mask

### DISCUSSION THREAD

**Discussion by njlewis** [saadeghi/daisyui#446] [Q&A] [answered]
Hi! I'm just getting started and used to using Tailwind CSS and have just discovered Daisy UI. Seems like a really useful addition to the toolbox. 

I am hoping to use a specific SVG shape as a mask element. I've found using Daisy UI e.g. `mask mask-circle` very straightforward but wanted to choose my own SVG file.

Is it possible to extend the library to include additional shapes i.e.:

 ```
mask: {
          'hex': 'url(../../images/hex.svg)',
      }
```

This is obviously quite simple to do using Tailwind CSS on it's own when extending to include new breakpoints, background images etc.

Any advice would be much appreciated!

**Comment by saadeghi** [comment, marked as answer]
Here's an example: https://play.tailwindcss.com/wl7OZ09Ka7?file=css

**Reply by njlewis** [reply]
Ah, brilliant! Incredibly easy then!

**Reply by Koaieus** [reply]
Is it also possible to define a custom class in terms of DaisyUI classes like this? 

```css
@tailwind base;
@tailwind components;
@tailwind utilities;

.border-custom {
  @apply border-base-content/20
}
```
Trying this gives the following error upon building:
```shell
ERROR: The border-base-content/20 class does not exist. If border-base-content/20 is a custom class, make sure it is defined within a @layer directive.
```
I tried more variations of the part in the css file after the three `@tailwind` directives, including:
```css
.border-custom {
  @apply border-base-content border-opacity-20
}
```
or 
```css
@layer base {
  .border-custom {
    @apply border-base-content border-opacity-20
  }
}
```
... but I keep getting the error about `border-base-content` not existing.
Is it possible to create classes like this based on existing DaisyUI classes?

**Reply by saadeghi** [reply]
@Koaieus 
It should work: https://play.tailwindcss.com/6duN4TZouX?file=css   
If it's not working, it's a problem with either your build tool or CSS processor or the framework that is processing your CSS.  

Maybe updating your dependencies to the latest version will help.

**Reply by Koaieus** [reply]
@saadeghi 
Thanks for the response, I will have to look into the processing! Currently working on a Nuxt3 monorepo and am trying to specify this custom class in a "base" `layer` from which all the `apps` in the monorepo are inheriting; it seems funky business is going on with the order in which tailwind (+DaisyUI plugin) is initialized/processed w.r.t. the child layers, I will have to do more research into what's going on exactly

**Reply by saadeghi** [reply]
@Koaieus let me know if I can help
